app/javascript/components/cloud-provider/validators.js

```diff
@@ -0,0 +1,29 @@
+
+const isHostname = function(value) {
+  return /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])$/.test(value);
```

Comment:
This doesn't look right.

You probably want something like `\p{L}` instead of `a-zA-Z`. Otherwise, this won't match any national domains...

```
> isHostname('♡.com')
false
```

Yet, http://♡.com exists and is a valid domain.

EDIT: the full list of valid unicode chars is probably here https://unicode.org/faq/idn.html